types/intset.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"github.com/Workiva/go-datastructures/bitarray"
+)
+
+// IntSet is a space-efficient set of uint64
+type IntSet struct {
+	ba bitarray.BitArray
+}
+
+// NewIntSet returns a new IntSet, optionally initialized with integers
+func NewIntSet(ints ...uint64) IntSet {
+	out := IntSet{ba: bitarray.NewSparseBitArray()}
+	for _, i := range ints {
+		out.Add(i)
+	}
+	return out
+}
+
+// Add adds an integer to this IntSet
+func (is IntSet) Add(i uint64) {
+	// Ignoring errors as we are using SparseBitArrays, which never return errors
+	_ = is.ba.SetBit(i)
+}
+
+// Contains returns whether an integer exists in this IntSet
+func (is IntSet) Contains(i uint64) bool {
+	// Ignoring errors as we are using SparseBitArrays, which never return errors
+	isSet, _ := is.ba.GetBit(i)
+	return isSet
+}
+
+// IsSubset returns true if every element in other is also in the receiver
+func (is IntSet) IsSubset(other IntSet) bool {
+	// The method we're calling here seems weird, but in bitarray (https://github.com/Workiva/go-datastructures/blob/f07cbe3f82ca2fd6e5ab94afce65fe43319f675f/bitarray/block.go#L97)
+	// "intersect" means, "wholly contained within"
+	return is.ba.Intersects(other.ba)
+}
+
+// Union returns a new IntSet, the result of a set union of the receiver and other
+func (is IntSet) Union(other IntSet) IntSet {
+	return IntSet{ba: is.ba.Or(other.ba)}
+}
```

Comment:
For conceptual consistency, I think the API offered here would be better if it were all in a mutable style, or all immutable. If our expected usage requires a mutable structure for performance, we can go that way. This would mean `AddAll`, `RetainAll` and `RemoveAll` instead of union, intersection and difference (and probably need a `Copy()` too). Else an immutable structure with methods like these.

This is not blocking feedback, use your discretion.

Reply:
Done, lets try out an immutable style